### PR TITLE
Add custom strategy tutorial

### PR DESCRIPTION
**Pull request by adam-narozniak**
Add custom strategy creation in jupyter tutorial

### CODE REVIEW

**Comment by danieljanes** [comment]
Is there a way to get a smaller diff?

**Comment by adam-narozniak** [comment]
Yes, sorry. My IDE automatically changed the indentations between the time I copied the version from google colab.

File: doc/source/tutorial/Flower-3-Building-a-Strategy-PyTorch.ipynb [hunk 1]
```diff
@@ -320,7 +320,7 @@
       "source": [
         "## Build a Strategy from scratch\n",
         "\n",
-        "[WIP - add description]"
+        "Let’s overwrite the configure_fit method such that it passes a higher learning rate (potentially also other hyperparameters to an optimizer) to a fraction of the client. We will keep the sampling of the clients in the standard way and then change the configuration dictionary (one of the FitIns elements)."
```
Comment (danieljanes) [inline doc/source/tutorial/Flower-3-Building-a-Strategy-PyTorch.ipynb:323]:
```suggestion
        "Let’s overwrite the `configure_fit` method such that it passes a higher learning rate (potentially also other hyperparameters) to the optimizer of a fraction of the clients. We will keep the sampling of the clients as it is in `FedAvg` and then change the configuration dictionary (one of the `FitIns` attributes)."
```

File: doc/source/tutorial/Flower-3-Building-a-Strategy-PyTorch.ipynb [hunk 2]
```diff
@@ -447,7 +514,7 @@
       "source": [
         "## Recap\n",
         "\n",
-        "[WIP - add description]"
+        "In this notebook, we’ve seen how to implement a custom strategy. It allows very flexible client creation, fitting, and evaluation. You have to overwrite the abstract methods of the Strategy class. And additionally, you can pass custom functions to your new class initializer to make it even more robust. "
```
Comment (danieljanes) [inline doc/source/tutorial/Flower-3-Building-a-Strategy-PyTorch.ipynb:517]:
```suggestion
        "In this notebook, we’ve seen how to implement a custom strategy. A custom strategy enables granular control over client node configuration, result aggregation, and more. To define a custom strategy, you only have to overwrite the abstract methods of the (abstract) base class `Strategy`. To make custom strategies even more powerful, you can pass custom functions to the constructor of your new class (`__init__`) and then call these functions whenever needed. "
```

File: doc/source/tutorial/Flower-3-Building-a-Strategy-PyTorch.ipynb [hunk 3]
```diff
@@ -399,18 +448,36 @@
...
+        "        \"\"\"Evaluate global model parameters using an evaluation function.\"\"\"\n",
         "\n",
-        "        # TODO WIP - add implementation\n",
+        "        # Let's assume we won't perform the global model evaluation on the server side.\n",
+        "        return None\n",
         "\n",
-        "        return None"
+        "    def num_fit_clients(self, num_available_clients: int) -> Tuple[int, int]:\n",
+        "        \"\"\"Return the sample size and the required number of available\n",
+        "        clients.\"\"\"\n",
```
Comment (danieljanes) [inline doc/source/tutorial/Flower-3-Building-a-Strategy-PyTorch.ipynb:473]:
```suggestion
        "        \"\"\"Return sample size and required number of clients.\"\"\"\n",
```

**Review by danieljanes — CHANGES_REQUESTED** [review]

**Review by danieljanes — APPROVED** [review]
Lgtm!

**Pull request merged**